src/ctr_logging.c

```diff
@@ -167,6 +167,13 @@ bool write_to_logs(stdpipe_t pipe, char *buf, ssize_t num_read)
  */
 int write_journald(int pipe, char *buf, ssize_t buflen)
 {
+	/* When using writev_buffer_append_segment, we should never approach the number of
+	 * entries necessary to flush the buffer. Therefore, the fd passed in is for /dev/null
+	 */
+	_cleanup_close_ int dev_null = open("/dev/null", O_WRONLY | O_CLOEXEC);
```

Comment:
now we open the the file each time `write_journald` is used.  Could we do it just once?

Reply:
we could, but that introduces the possibility of leaks, as it'd have to be a global (or we'd have to significantly rework the api structure). Do you think those options are worth it?

Reply:
opening it once and leaking it should be fine I think.  We could leak it in the function without affecting the api.